src/main/java/services/DBService.java

```diff
@@ -37,4 +39,31 @@ public User getUserDetails(int userID) throws ExecutionException, InterruptedExc
 
     }
 
-}
+    public void updateDefaultLang(User user, String newDefaultLang) {
+        Firestore dbFirestore = FirestoreClient.getFirestore();
+        // System.out.println(user.getDefaultLang());
+        String docInfo = "id" + user.getUser_id();
+        user.setDefaultLang(newDefaultLang);
+        DocumentReference docRef = dbFirestore.collection("users").document(docInfo);
+        ApiFuture<WriteResult> future = docRef.update("users", user.getDefaultLang());
+        // System.out.println(user.getDefaultLang());
+    }
+    public void updateName(User user, String name) {
+        Firestore dbFirestore = FirestoreClient.getFirestore();
+        // System.out.println(user.getName());
+        String docInfo = "id" + user.getUser_id();
+        user.setName(name);
+        DocumentReference docRef = dbFirestore.collection("users").document(docInfo);
+        ApiFuture<WriteResult> future = docRef.update("users", user.getName());
```

Comment:
Instead of ApiFuture<WriteResult> future = docRef.update("users", user.getName()); on line 57, have:

ApiFuture<WriteResult> future = docRef.update("name", user.getName());